speechbrain/lobes/models/transformer/TransformerASR.py

```diff
@@ -488,15 +515,20 @@ def encode(
             src = src + self.positional_encoding(src)
             pos_embs_source = None
 
-        encoder_out, _ = self.encoder(
+        outputs = self.encoder(
             src=src,
             src_mask=src_mask,
             src_key_padding_mask=src_key_padding_mask,
             pos_embs=pos_embs_source,
             dynchunktrain_config=dynchunktrain_config,
         )
 
-        return encoder_out
+        if self.output_hidden_states:
+            encoder_out, _, hidden_states = outputs
+            return encoder_out, hidden_states
+        else:
+            encoder_out, _ = outputs
+            return encoder_out
```

Comment:
IMO there should be tests for the `output_hidden_states` which would be covering all code paths (i.e. testing with different encoders) and test for the length of the array. The control flow here is a bit messy and prone to breaking when changing the nearby code in the future.

Reply:
I agree... but I somehow don't want to do this now... what do you suggest?

Reply:
Procrastinate, yolo